lib/experimental/html/class-wp-html-tag-processor.php

```diff
@@ -1113,20 +1113,20 @@ private function after_tag() {
 	}
 
 	/**
-	 * Converts class name updates into tag attributes updates
+	 * Converts class name updates into tag lexical updates
 	 * (they are accumulated in different data formats for performance).
 	 *
-	 * This method is only meant to run right before the attribute updates are applied.
+	 * This method is only meant to run right before the lexical updates are applied.
 	 * The behavior in all other cases is undefined.
 	 *
 	 * @return void
 	 * @since 6.2.0
 	 *
 	 * @see $classname_updates
-	 * @see $attribute_updates
+	 * @see $lexical_updates
 	 */
-	private function class_name_updates_to_attributes_updates() {
-		if ( count( $this->classname_updates ) === 0 || isset( $this->attribute_updates['class'] ) ) {
+	private function class_name_updates_to_lexical_updates() {
```

Comment:
this function still terminates in a call to `set_attribute()` or `remove_attribute()` so I'm struggling with the rename

Reply:
I thought it was consistent, since `class_name_updates_to_lexical_updates` does indeed take all items from `$this->classname_updates` and moves them into (an item with key `class` of) `$those->$lexical_updates`. So the name of the `private` method kind of reflects what happens to those `private` members; it just happens to be done via a `public` method.

Arguably, this is another result from renaming the private field in preparation of allowing it to hold non-attribute updates, when it doesn't do that just yet -- as is reflected by its public methods (which are still `set_attribute` and `get_attribute`).

Ultimately, I'd lean towards keeping `class_name_updates_to_lexical_updates` here (mostly because of the "private method's name reflects the two private members it operates on" argument).

Reply:
like the changed comment before, I feel like these changes are focused on the internals of how things end up being instead of what their intentions are. yes it's true that these end up in a lexical update structure, but my point in mentioning `set_attribute`/`remove_attribute` is that this is exactly what the function is doing: it's taking a set of class name changes and collapsing those into a single attribute update.

if somehow we enqueued another attribute update to `class` it would override these, though I guess we should ensure that doesn't happen, lest we repeat the same overarching problem of write-read cycles and needing to modify enqueued updates.

for what it's worth my original intention was simply to rename `$this->attribute_updates` to `$this->lexical_updates` which is now close to the state of the PR. it wasn't so much about private/public interface names but more about naming things internally in line with how they're used.

this function at least is still used to turn class name updates into an attribute update

Reply:
Fair enough, I don't mind making this PR solely rename `$this->attribute_updates` to `$this->lexical_updates`; it was out of my striving for completeness that I ended up renaming some related fields as well. But I agree that it's not as cut-and-dry for some of them, so I'm happy to revert.

Reply:
b24d6c7bbda7182fce7c8ccbd01fa9f846ebfd34